testrunner-running/src/main/scala/edu/illinois/cs/statecapture/StateCapture.java

```diff
@@ -0,0 +1,489 @@
+                            UnmarshalChain.reset();
+                            UnmarshalChain.initializeChain(fieldList[i].getDeclaringClass().getName(), fieldList[i].getName());
+                            obj = xstream.fromXML(state0);
+                        }
+                        if (obj == null) {
+                            System.out.println("Unable to construct the relevant object during load");
+                            return;
+                        }
+                        if (AccessibleObject.class.isAssignableFrom(obj.getClass())) {
+                            ((AccessibleObject) obj).setAccessible(true);
+                        }
+                        // the purpose is to serialize/deserialize the object wrapped within a ThreadLocal.
+                        if (threadLocal) {
+                            Object tmp = fieldList[i].get(null);
+                            ((ThreadLocal)tmp).set(obj);
+                            obj = tmp;
+                        }
+
+                        if (!Mockito.mockingDetails(obj).isMock()) {
+                            FieldUtils.writeField(fieldList[i], (Object) null, obj, true);
+                        }
+
+                        String output = fieldName + " set\n";
+                        Files.write(Paths.get(reflectionFile), output.getBytes(),
+                                StandardOpenOption.APPEND);
+                    } catch (IllegalArgumentException | IllegalAccessException | NoSuchMethodException
+                            | SecurityException | InvocationTargetException e) {
+                        e.printStackTrace();
+                        String outputNormalError = fieldName + " reflectionError: " + e + "\n";
+                        Files.write(Paths.get(reflectionFile), outputNormalError.getBytes(),
+                                StandardOpenOption.APPEND);
+                    }
+                }
+            } catch (ClassNotFoundException | SecurityException e) {
+                e.printStackTrace();
+                String output = fieldName + " deserializeError: " + e + "\n";
+                Files.write(Paths.get(reflectionFile), output.getBytes(),
+                        StandardOpenOption.APPEND);
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+            String output = fieldName + " deserializeError: " + e + "\n";
+            Files.write(Paths.get(reflectionFile), output.getBytes(),
+                    StandardOpenOption.APPEND);
+        }
+    }
+
+    @Override
+    public void capture() {
+        try {
+            if (xmlDir.isEmpty() || rootFile.isEmpty()) {
+                System.out.println("WARNING: The xml directory or root file are not provided, thus it will not do capturing.");
+                return;
+            }
+            String state = Configuration.config().getProperty("statecapture.state", "");
```

Comment:
If this is the only way `statecapture.state` is being used, maybe we can specify it to only be relevant for controlling whether eagerload should be done or not.